src/cargo/util/lints.rs

```diff
@@ -191,43 +190,33 @@ fn verify_feature_enabled(
 }
 
 pub fn get_span(
-    document: &Document<String>,
+    document: &toml::Spanned<toml::de::DeTable<'static>>,
     path: &[&str],
     get_value: bool,
 ) -> Option<Range<usize>> {
-    let mut table = document.as_item().as_table_like()?;
+    let mut table = document.get_ref();
     let mut iter = path.into_iter().peekable();
     while let Some(key) = iter.next() {
-        let (key, item) = table.get_key_value(key)?;
+        let key_s: &str = key.as_ref();
+        let (key, item) = table.get_key_value(key_s)?;
         if iter.peek().is_none() {
             return if get_value {
-                item.span()
+                Some(item.span())
             } else {
-                let leaf_decor = key.dotted_decor();
-                let leaf_prefix_span = leaf_decor.prefix().and_then(|p| p.span());
-                let leaf_suffix_span = leaf_decor.suffix().and_then(|s| s.span());
-                if let (Some(leaf_prefix_span), Some(leaf_suffix_span)) =
-                    (leaf_prefix_span, leaf_suffix_span)
-                {
-                    Some(leaf_prefix_span.start..leaf_suffix_span.end)
-                } else {
-                    key.span()
-                }
+                Some(key.span())
```

Comment:
So the span method is effectively the same as the previous implementation that includes decorations?
(Cannot tell unless I look into the implementation)

Reply:
We were getting the span of the entire header by grabbing the spans of the decor and making a span of what came between.  Looking at https://github.com/rust-lang/cargo/pull/15736/files/539a48452a03a68d18cd32481e364550ee42f329#diff-1c71ec9c445774ebb8b0c3dd4c35817dda698c78e15fb83d8a61d8b145c5dd34, it appears that that behavior is now the default for spans so this is built-in now.  Even if it wasn't, its likely not worth it to keep using `toml_edit`.